tools/release/adot-operator-images-mirror/get-operator-tag.sh

```diff
@@ -0,0 +1,27 @@
+#! /bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+COLLECTOR_TAG=$(cat go.mod | grep "go.opentelemetry.io/collector " | cut -d " " -f2)
+COLLECTOR_TAG_WITHOUT_PATCH_VER=$(echo $COLLECTOR_TAG | sed 's/\..$//')
+
+OPERATOR_TAGS=$(echo $(curl https://api.github.com/repos/open-telemetry/opentelemetry-operator/tags) | jq  '.[] | .name')
+OPERATOR_TAG=$(echo "$OPERATOR_TAGS" | grep "$COLLECTOR_TAG_WITHOUT_PATCH_VER" -m 1 | sed 's/"//g')
```

Comment:
```suggestion
OPERATOR_TAG=$(curl https://api.github.com/repos/open-telemetry/opentelemetry-operator/tags | jq  "map(select( .name | startswith(\"${COLLECTOR_TAG_WITHOUT_PATCH_VER}\"))) | first | .name")
```

Let `jq` do the heavy lifting for you.  I'm not 100% certain the escaping on this is correct, but the `jq` pipeline works.  It `map`s each object in the array, `select`ing those whose `.name` `startswith` the desired value, then takes the `first` item in the resulting array and extracts its `.name`.